src/platform/Zephyr/BLEManagerImpl.cpp

```diff
@@ -563,6 +583,37 @@ CHIP_ERROR BLEManagerImpl::HandleTXCharComplete(const ChipDeviceEvent * event)
     return CHIP_NO_ERROR;
 }
 
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+CHIP_ERROR BLEManagerImpl::PrepareC3CharData()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    char serialNumber[ConfigurationManager::kMaxSerialNumberLength + 1];
+    uint16_t lifetimeCounter = 0;
+    BitFlags<AdditionalDataFields> additionalDataFields;
+
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
```

Comment:
if it's not set you would pass uninitialized `serialNumber` to `generateAdditionalDataPayload`. Either always call `GetSerialNumber` and `GetLifetimeCounter` or initialize the array with `= {}`.

Reply:
Yeah, but `serialNumber` passed to method will not be used in that case, as rotating device id flag will not be set here: https://github.com/kkasperczyk-no/connectedhomeip/pull/52/files#diff-5d1fa6246af82f5f83ad58a70c7eb429c89400997eff8f7d18dd4b6a96acc766R601. Nevertheless it's true that maybe trusting underlaying implementation is not that safe.